web/src/engine/websites/ColoredManga.ts

```diff
@@ -0,0 +1,98 @@
+import { Tags } from '../Tags';
+import icon from './ColoredManga.webp';
+import { Chapter, DecoratableMangaScraper, Manga, type MangaPlugin, Page } from '../providers/MangaPlugin';
+import * as Common from './decorators/Common';
+import { FetchJSON } from '../platform/FetchProvider';
+import type { Priority } from '../taskpool/DeferredTask';
+
+type APIManga = {
+    name: string
+    chapters: APIChapter[]
+}
+
+type APIChapter = {
+    id: string,
+    title: string,
+    number: string,
+    totalImage: number
+}
+
+type APIImage = {
+    image : string
+}
+export default class extends DecoratableMangaScraper {
+    private readonly apiUrl = `${this.URI.origin}/api/`;
+    private readonly mangaRegexp = new RegExp(`^${this.URI.origin}/manga/([^/]+)$`);
+
+    public constructor() {
+        super('coloredmanga', 'Colored Manga', 'https://coloredmanga.net', Tags.Media.Manhwa, Tags.Media.Manga, Tags.Language.English, Tags.Source.Scanlator);
+    }
+
+    public override get Icon() {
+        return icon;
+    }
+
+    public override ValidateMangaURL(url: string): boolean {
+        return this.mangaRegexp.test(url);
+    }
+
+    public override async FetchManga(provider: MangaPlugin, url: string): Promise<Manga> {
+        const mangaSlug = url.match(this.mangaRegexp)[1];
+        const mangaData = await this.GetMangaData(mangaSlug);
+        return new Manga(this, provider, mangaSlug, mangaData.name);
+    }
+
+    public override async FetchMangas(provider: MangaPlugin): Promise<Manga[]> {
+        const mangas = await Common.FetchMangasSinglePageCSS.call(this, provider, '/manga', 'div#themes_outside__WCut6 a:not([id])');
+        return mangas.map(manga => new Manga(this, provider, manga.Identifier.match(/\/manga\/([^/]+)/)[1], manga.Title));
+    }
+
+    public override async FetchChapters(manga: Manga): Promise<Chapter[]> {
+        const { chapters } = await this.GetMangaData(manga.Identifier);
+        return chapters.map(chapter => {
+            const title = [
+                chapter.number,
+                chapter.title ? '-' : '',
+                chapter.title,
+            ].join(' ').trim();
+            return new Chapter(this, manga, chapter.id, title);
+        });
+    }
+
+    public override async FetchPages(chapter: Chapter): Promise<Page[]> {
+        const { name, chapters } = await this.GetMangaData(chapter.Parent.Identifier);
+        const url = new URL(`/images/content/${name}/${chapter.Title}`, this.URI);
```

Comment:
Composing an URL to get the encoded pathname just to later decode the pathname again seems redundant.
Consider using the pathname directly:
```javascript
const path = `/images/content/${name}/${chapter.Title}`;
```